src/lib/common/string.cpp

```diff
@@ -213,6 +214,63 @@ int stringSplit(const std::string& in, char delimiter, std::vector<std::string>&
 
 
 
+/* *****************************************************************************
+*
+* hostnameIsValid - check a hostname for validity
+*
+* See https://en.wikipedia.org/wiki/Hostname
+*/
+static bool hostnameIsValid(const char* hostname, bool isIPv6)
+{
+  if (isIPv6 == true)  // No checks for IPv6
+  {
+    // FIXME P4: Implement hostnameIsValid for IPv6 addresses
```

Comment:
Hostnames are the same for IPv4 or IPv6. It is the name resolver (e.g. `/etc/hosts`, the DNS subsystem) the one which will provide either a IPv4 or IPv6 (or even both... I don't remember right now the details).

Thus, is this is an actual hostname checking function, I think whe whole IPV6 hasn't nothing to do.